microprofile/fault-tolerance/src/test/java/io/helidon/microprofile/faulttolerance/AsynchronousTest.java

```diff
@@ -123,4 +122,12 @@ void testAsyncCompletableFutureWithFallbackFailure() throws Exception {
         assertThat(completableFuture.get(), is("fallback"));
         assertThat(bean.wasCalled(), is(true));
     }
+
+    @Test
+    void testAsyncWithExecutor() throws Exception {
+        assertThat(bean.wasCalled(), is(false));
+        CompletableFuture<String> future = bean.asyncWithExecutor();
+        future.get();
```

Comment:
timeout

Reply:
All these tests need to be "fixed" in that way. The implementation "should" never block that way.